Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -612,6 +618,54 @@ public class GraphTraverser: GraphTraversing {
         return references
     }
 
+    /// Recursively find platform filters within transitive dependencies
+    /// - Parameters:
+    ///   - rootDependency: dependency whose platform filters we need when depending on `transitiveDependency`
+    ///   - transitiveDependency: target dependency
+    /// - Returns: PlatformFilters to apply to transitive dependency or `nil` if there isnt a path or path to a dependency results
+    /// in a disjoint
+    /// set of platform filters
+    func platformFilters(to transitiveDependency: GraphDependency, from rootDependency: GraphDependency) -> PlatformFilters {
```

Comment:
perhaps we have a dedicated result type here to help disambiguate no filters / `.disjoint` / filters - re-using the `PlatformFilters` is a bit tricky as in some traversals some dependencies have no filters and that's getting flagged as `.invalid`  :/

Reply:
The hope was to actually ensure that `[]` *is* invalid.  It makes little sense to have a dependency (or other type) that doesnt apply to any platform.  So the goal is going to be to actually make that be the case as we resolve our filters.